packages/app/src/app/components/sandbox/Preview/DevTools/Console/Input.js

```diff
@@ -1,95 +1,188 @@
+/* @flow */
 import React from 'react';
 import styled from 'styled-components';
 
 import ChevronRight from 'react-icons/lib/md/chevron-right';
 import theme from 'common/theme';
 
+import MonacoEditor from 'app/components/sandbox/CodeEditor/monaco/MonacoReactComponent';
+
 import { IconContainer } from './styles';
 
+const CONSOLE_INPUT_TOP_PADDING = 6;
+const CONSOLE_INPUT_BOTTOM_PADDING = 6;
+const CONSOLE_INPUT_PADDING =
+  CONSOLE_INPUT_TOP_PADDING + CONSOLE_INPUT_BOTTOM_PADDING;
+const CONSOLE_INPUT_LINE_HEIGHT = 20;
+const CONSOLE_INPUT_MAX_HEIGHT = 110;
+
 const Container = styled.div`
+  flex-shrink: 0;
   position: relative;
-  height: 2rem;
+  height: ${props => props.height}px;
   min-height: 2rem;
   width: 100%;
   background-color: ${props => props.theme.background.darken(0.3)};
-
   display: flex;
-  align-items: center;
+  align-items: flex-start;
+
+  .monaco-editor-background {
+    background-color: ${props => props.theme.background.darken(0.3)};
+  }
 `;
 
-const Input = styled.input`
+type Props = {
+  evaluateConsole: (command: string) => void,
+};
+
+const InputWrapper = styled.div`
   position: relative;
-  height: 1.5rem;
+  height: 100%;
+  padding-top: ${CONSOLE_INPUT_TOP_PADDING}px;
+  padding-bottom: ${CONSOLE_INPUT_BOTTOM_PADDING}px;
+  box-sizing: border-box;
   width: 100%;
-  background-color: ${props => props.theme.background.darken(0.3)};
-  border: none;
-  outline: none;
-  color: rgba(255, 255, 255, 0.8);
-  font-family: Menlo, monospace;
-  font-size: 13px;
 `;
 
-type Props = {
-  evaluateConsole: (command: string) => void,
+const monacoOptions = {
+  // language: 'javascript',
```

Comment:
Why did you comment this line?

Reply:
Thanks @CompuIves ! Pleasure to help. 

Do you want syntax highlighting and autosuggestion as well 😃 ?
![syntax-highlight](https://user-images.githubusercontent.com/679275/34924159-9b519fd0-f956-11e7-85bd-b09af110ba7d.gif)

But for that we would need to detect if we are in CodeMirror mode.ecause it looks that in Monaco there's only one theme for editors. (https://github.com/Microsoft/monaco-editor/issues/338)

Reply:
The above changes (extracted to a separate file): https://github.com/xyc/codesandbox-client/commit/8b4f76d099b65080c80bf66acc32867f58c89825. Please let me know what you think @CompuIves

Reply:
> Edit: Alternative approach is to extract defineTheme logic from Monaco.js into MonacoReactComponent/a separate file and have a guard that prevents the theme from being defined twice. That way even in code mirror mode we shall still have the theme available. (defining theme on input component mounting will sometimes result in text re-colorized briefly in other Monaco editors so it's best that theme is only defined once).

I like this approach, sounds good!

Syntax Highlighting and autosuggest are super. I love to work with this console now!